examples/cifar10/train.cpp

```diff
@@ -104,21 +116,75 @@ void train_cifar10(string data_dir_path, double learning_rate, ostream &log) {
                           n_train_epochs, on_enumerate_minibatch,
                           on_enumerate_epoch);
 
-  cout << "end training." << endl;
+  std::cout << "end training." << std::endl;
 
   // test and show results
-  nn.test(test_images, test_labels).print_detail(cout);
-
+  nn.test(test_images, test_labels).print_detail(std::cout);
   // save networks
-  ofstream ofs("cifar-weights");
+  std::ofstream ofs("cifar-weights");
   ofs << nn;
 }
 
+static core::backend_t parse_backend_name(const char *name) {
+  static const char *names[] = {
```

Comment:
@edgarriba `std::basic_string::cpmpare` is case sensitive and `strcasecmp` function ignores case so that I used that function.

You don't need to (actually, you shouldn't) use `std::string` just to wrap string literals.
I hope something similar to `boost::string_ref` would appear in future C++ standard though.